srml/council/src/lib.rs

```diff
@@ -14,7 +14,243 @@
+//! that the operation will not require substantial computation. The voters' account id is added to the list `Voters`
+//! of present voters.
+//!
+//! - **Candidate voters' subsequent votes (after their first vote)** If the voter makes a subsequent vote that's
+//! valid, then their vote is recorded in `LastActiveOf`, which maps their account id to the last cleared vote index
+//! that they were active at, and the votes (i.e. yay or nay) for each candidate with a vote index are added to the
+//! `ApprovalsOf` mapping.
+//!
+//! - **Candidate voter inactivity reaping process** After determining the claims' validity, we call `remove_voter`
+//! as follows depending on the claim validity: if the claim was valid delete the inactive voter, otherwise delete
+//! the reporter. Lastly we perform reaping as follows, depending on the claim validity: if the claim is valid call
+//! `repatriate_reserved` to slash the target account of their voter bond and move that value from the targets'
+//! reserved account balance to the free balance of the reporter (beneficiary) and emit a `VoterReaped` event,
+//! otherwise if the claim was invalid call `slash_reserved` to slash the reporter for their bad behaviour in making a
+//! false claim. The reporter's account is slashed by deducting a value from their reserved balance, and by decreasing
+//! the total amount of stake in the system by the amount that was slashed, then we emit a `BadReaperSlashed` event.
+//!
+//! ### Goals
+//!
+//! The Council module in Substrate is designed to make the following possible:
+//!
+//! - Creation of council proposals by councillors using the council motion mechanism.
+//! - Validation of council proposals.
+//! - Tallying votes of council proposals by councillors during the proposals' voting period.
+//! - Vetoing (postponement) of council proposals for a cooldown period through abstention by councillors.
+//! - Elevation of council proposals to start an associated referenda on the table of referenda.
+//! - Applying vote thresholds to referenda depending on their associated council proposal voting approval tally.
+//! - Instant autonomous execution of referenda once their vote tally reaches the vote threshold level of approval.
+//! - Cancellation of council proposals that were elevated as associated referenda onto the table of referenda.
+//! - Candidate registration in list slots (necessary to receive candidate approval votes during the
+//! presentation period).
+//! - Deduction of security bonds from candidate voters.
+//! - Express council seat candidate approval voting.
+//! - Reaping of candidate voters due to valid claims of their inactivity by reporters.
+//! - Reaping reporters that lodge invalid claims of candidate voter inactivity.
+//!
+//! ## Interface
+//!
+//! ### Dispatchable Functions
+//!
+//! #### Council Motions
+//!
+//! - `propose` - Create a council proposal using the council motion mechanism by a councillor who provides a vote
```

Comment:
Should these be wrapped?

Reply:
i've adopted the approach suggested by @joepetrowski where we're using vertical rulers.
in VSCode i've changed settings.json to be the following. The file is located at ~/.config/Code/User/settings.json.
we just make sure the width of the text doesn't exceed 120 chars

```
{
    "editor.rulers": [
        80,
        120
    ],
    "editor.tabSize": 2
}
```

Reply:
![Screenshot from 2019-04-26 11-11-34](https://user-images.githubusercontent.com/6226175/56797715-97afe600-6815-11e9-841b-2fe9140d4fa3.png)

Reply:
^Yeah, anything that falls within 120, especially for these more verbose docs, is OK.